src/cascadia/TerminalApp/App.cpp

```diff
@@ -215,11 +215,49 @@ namespace winrt::TerminalApp::implementation
     void App::_CreateNewTabFlyout()
     {
         auto newTabFlyout = Controls::MenuFlyout{};
+        auto _keyBindings = _settings->GetKeybindings();
+
         for (int profileIndex = 0; profileIndex < _settings->GetProfiles().size(); profileIndex++)
         {
             const auto& profile = _settings->GetProfiles()[profileIndex];
             auto profileMenuItem = Controls::MenuFlyoutItem{};
 
+            // add the keyboard shortcuts for the first 9 profiles
+            if (profileIndex < 9)
+            {
+                // enum value for ShortcutAction::NewTabProfileX starts at 3; 0==3
+                auto profileKeyChord = _keyBindings.GetKeyBinding(static_cast<ShortcutAction>(profileIndex + 3));
+                
+                // make sure we find one to display
+                if (profileKeyChord)
+                {
+                    // work around https://github.com/microsoft/microsoft-ui-xaml/issues/708 in case of VK_OEM_COMMA
+                    if (profileKeyChord.Vkey() != VK_OEM_COMMA)
+                    {
+                        // use the XAML shortcut to give us the automatic capabilities
+                        auto profileShortcut = Windows::UI::Xaml::Input::KeyboardAccelerator{};
+
+                        // TODO: Modify this when https://github.com/microsoft/terminal/issues/877 is resolved
+                        profileShortcut.Key(static_cast<Windows::System::VirtualKey>(profileKeyChord.Vkey()));
+
+                        // inspect the modifiers fromt he KeyChord and set the flags int he XAML value

```

Comment:
```suggestion
                        // inspect the modifiers from the KeyChord and set the flags in the XAML value
```

Reply:
Resolving in next update (found another instance)